app/views/dashboard/autoyast.xml.erb

```diff
@@ -24,6 +24,18 @@
         ]]>
         </source>
       </script>
+      <script>
+        <chrooted config:type="boolean">true</chrooted>
+        <filename>set_authorized_keys.sh</filename>
+        <interpreter>shell</interpreter>
+        <source><![CDATA[
+        #!/bin/sh
+        mkdir -p /root/.ssh
+        chmod 600 /root/.ssh
+        echo "<%= @ssh_public_key %>" > /root/.ssh/authorized_keys
+        ]]>
+        </source>
+      </script>
```

Comment:
I think we can avoid writing this at all if we haven't a ssh key, so we could do something like this:

```
<% unless @ssh_public_key.blank? %>
      <script>
        <chrooted config:type="boolean">true</chrooted>
        <filename>set_authorized_keys.sh</filename>
        <interpreter>shell</interpreter>
        <source><![CDATA[
        #!/bin/sh
        mkdir -p /root/.ssh
        chmod 600 /root/.ssh
        echo "<%= @ssh_public_key %>" > /root/.ssh/authorized_keys
        ]]>
        </source>
      </script>
<% end %>
```